ru/turing/courses/lesson3/filippov/Registry.java

```diff
@@ -0,0 +1,102 @@
+package ru.turing.courses.lesson3.filippov;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+public class Registry<T extends Record> {
+    private final Map<String, T> storage = new HashMap<>();
+
+    //add(key, value) - добавить в реестр новую запись
+
+    public void add(String key, T value) {
+        storage.put(value.getId(), value);
+    }
+
+    //removeByKey(key) - удалить из реестра запись по заданному ключу. Возвращать должен значение записи
+
+    public void removeByKey(String key) {
+
+        if (storage.containsKey(key)) {
+            T removeRecord = storage.get(key);
+            storage.remove(key);
+            System.out.println("Был удален пользователь:\n" + removeRecord.getRecord());
+        }
+        else {
+            System.out.println("Пользователя с таким ключом нет(");
+        }
+    }
+
+    //- removeByValue(value) - удалить из реестра все записи по заданному значению. Возвращать должен List ключей удаленных записей
+
+    public List removeByValue(T value) {
```

Comment:
```suggestion
    public List<String> removeByValue(T value) {
```
Ты знаешь тип ключа, почему бы не указать его явно?